📖 Documentation for go/v2 and go/v3 plugin (#2569)
kubernetes-sigs/kubebuilder: docs/book/src/plugins/go-v2-plugin.md

# go/v2 plugin

> This plugin is also used to build operators similar to [go/v3](go-v3-plugin.md). Check the following [example](https://github.com/kubernetes-sigs/kubebuilder/tree/master/testdata/project-v2) to view the files created or updated by this plugin when used with the supported subcommands.
> 

Reviewer: ```suggestion
The go/v2 plugins have the purpose to scaffold Golang projects and keep the backwards compatibility with the default scaffold made using kubebuilder CLI 2.x.z releases.   

<node>
You can check samples using this plugin by looking at the `project-v2-<options>` samples under the [testdata][testdata] directory on the root directory of the Kubebuilder project.
</node>  

```

> ## How to use it ?

Reviewer: ```suggestion
## When should I use this plugin

Only if you are looking to scaffold a project with the legacy layout. Otherwise, it is recommended you to use the default Golang version plugin. 

<aside class="note warning">

Be aware that this plugin version does not provide a scaffold compatible with the latest versions of the dependencies used in order to keep its backwards compatibility. 

</aside>

## How to use this plugin ?
```

> This plugin should only be used to scaffold Go projects which are not backwards compatible with the Kubebuilder layout >= 2.0.0 and < 3.0.0. 
> Use the following command to create a new project with the go/v2 plugin:

Reviewer: ```suggestion
To initialize a Golang project using the legacy layout and with this plugin run: 
```

```sh
kubebuilder init --domain tutorial.kubebuilder.io --repo tutorial.kubebuilder.io/project --plugins=go/v2
```

> All the other subcommands supported by go/v2 plugin can be executed similarly.

Reviewer: ```suggestion
<aside>

By creating a project with this plugin, the PROJECT file scaffold will be using the previous schema (_project version 2_).  So that Kubebuilder CLI knows what plugin version was used and will call its subcommands such as `create api` and `create webhooks`.  Note that further Golang plugins versions use the new Project file schema, which tracks the information about what plugins and versions have been used so far. 

</aside>
```

## Subcommands supported by the plugin

> - [x] Init

Reviewer: ```suggestion
- Init ($ kubebuilder init [OPTIONS])
```

I think we do not need the x. Also, could we supplement the follow ones? WDYT?

Author: Ya I think that would be better. I will make the changes

# go/v2 plugin

The `go/v2` plugin has the purpose to scaffold Golang projects to help users to build projects with [controllers][controller-runtime] and keep the backwards compatibility with the default scaffold made using Kubebuilder CLI `2.x.z` releases.   

<node>

You can check samples using this plugin by looking at the `project-v2-<options>` directories under the [testdata][testdata] projects on the root directory of the Kubebuilder project.

</node>  

## When should I use this plugin

Only if you are looking to scaffold a project with the legacy layout. Otherwise, it is recommended you to use the default Golang version plugin. 

<aside class="note warning">

> 

Reviewer: ```suggestion
<h1> Supportability </h1>
```

Be aware that this plugin version does not provide a scaffold compatible with the latest versions of the dependencies used in order to keep its backwards compatibility. 

</aside>

## How to use it ?

To initialize a Golang project using the legacy layout and with this plugin run, e.g.:

```sh
kubebuilder init --domain tutorial.kubebuilder.io --repo tutorial.kubebuilder.io/project --plugins=go/v2
```
<aside class="note">

> 

Reviewer: ```suggestion
<h1> Note </h1>
```
We meed to add the title for it render right

# go/v2 plugin

The `go/v2` plugin has the purpose to scaffold Golang projects to help users to build projects with [controllers][controller-runtime] and keep the backwards compatibility with the default scaffold made using Kubebuilder CLI `2.x.z` releases.   

<node>

> You can check samples using this plugin by looking at the `project-v2-<options>` samples under the [testdata][testdata] directory on the root directory of the Kubebuilder project.

Reviewer: ```suggestion
You can check samples using this plugin by looking at the `project-v2-<options>` directories under the [testdata][testdata] directory on the root directory of the Kubebuilder project.
```

<aside class="note warning">

Be aware that this plugin version does not provide a scaffold compatible with the latest versions of the dependencies used in order to keep its backwards compatibility. 

</aside>

## How to use it ?

To initialize a Golang project using the legacy layout and with this plugin run, e.g.:

```sh
kubebuilder init --domain tutorial.kubebuilder.io --repo tutorial.kubebuilder.io/project --plugins=go/v2
```
<aside class="note">

By creating a project with this plugin, the PROJECT file scaffold will be using the previous schema (_project version 2_).  So that Kubebuilder CLI knows what plugin version was used and will call its subcommands such as `create api` and `create webhooks`.  Note that further Golang plugins versions use the new Project file schema, which tracks the information about what plugins and versions have been used so far. 

</aside>

## Subcommands supported by the plugin

-  Init -  `kubebuilder init [OPTIONS]`
-  Edit -  `kubebuilder edit [OPTIONS]`
-  Create API -  `kubebuilder create api [OPTIONS]`
-  Create Webhook - `kubebuilder create webhook [OPTIONS]`

> 

Reviewer: ```suggestion
```
nit unnecessary space

# go/v2 plugin

> The `go/v2` plugin has the purpose to scaffold Golang projects to help users to build projects with [controllers][controller-runtime] and keep the backwards compatibility with the default scaffold made using Kubebuilder CLI `2.x.z` releases.   

Reviewer 2: "...to scaffold Golang projects to help users..." -> "... to scaffold Golang projects by helping users..."

Reviewer 2: "...to scaffold..." -> "...of scaffolding..."

<node>

You can check samples using this plugin by looking at the `project-v2-<options>` directories under the [testdata][testdata] projects on the root directory of the Kubebuilder project.

</node>  

> ## When should I use this plugin

Reviewer 2: add a question mark
